### DISCUSSION TITLE
Implement onPoiClick for Apple Maps

### DISCUSSION THREAD

**Discussion by wkev-n** [react-native-maps/react-native-maps#4729] [Feature requests]
The release of iOS 16 introduced [selectable map features](https://developer.apple.com/documentation/mapkit/mkmapview/3975750-selectablemapfeatures?language=objc) to Apple Maps, allowing users to select points of interest. However, there is currently no callback implemented for this feature in `react-native-maps`.

### Proposal

To add support for selectable map features, I'm proposing reusing `onPoiClick` in `MapView` that is triggered when a user selects a point of interest. This function should behave similarly to `onPoiClick()` for Google Maps, but with a different type definition triggered since Apple Maps places do not have place IDs. We can populate it with data in [`MKPlacemark`](https://developer.apple.com/documentation/mapkit/mkplacemark).

#### Possible Implementation

We'll first need to configure `MapView` to have selectable map features:

```
map.selectableMapFeatures = 1 << MKMapFeatureTypePointOfInterest;
```

The proposed method definition for the Apple Maps version of `onPoiClick()` would be something like this:

```
export type AppleMapsPoiClickEvent = NativeSyntheticEvent<{
    name: string;
    location: LatLng;
    address: string;
}>;
```
By default, Apple animates selected points of interests. We can avoid that to by deselecting them in the `didSelectAnnotationView` callback.

**Comment by wkev-n** [comment]
I have a working version that I'm using as a patch currently. How does the team triage incoming pull requests?

**Reply by a-j-bailey** [reply]
Hey! Curious about the working implementation you have! Could you push it to a repo (or link a PR if you have it) so I can take a look at what you did to make it work? Trying to do the same.

**Reply by wkev-n** [reply]
hey @a-j-bailey! sorry for the late reply here. here's a patch file that works for my purposes for react-native-maps v1.7.1: https://gist.github.com/wkev-n/b217a0c7a8a091a3222171335fda4dc1. hope it helps!

**Reply by a-j-bailey** [reply]
Dude thank you! 
I'll take a closer look later but this seems super helpful.

**Comment by haardes** [comment]
Any news regarding this feature? Much needed.

**Reply by wkev-n** [reply]
No response from the team but I'm also no longer using it for my work. The patch in the thread above worked while I needed it and works well enough for an experiment if you're interested in integrating for yourself.

**Comment by jamesedmonston** [comment]
For anybody needing this on `1.24.3`, here's a patch created with Bun: https://gist.github.com/jamesedmonston/d78258f928e6307859415da81fd6012b

Thanks @wkev-n for your initial patch, which got me most of the way there!

**Reply by wkev-n** [reply]
Glad it was helpful!
